eo-parser/src/main/resources/org/eolang/parser/errors/rt-and-atoms.xsl

```diff
@@ -22,20 +22,32 @@ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 SOFTWARE.
 -->
-<!--
-  @todo #2921:30m Add check if there are atoms, but no +rt meta. It should be illegal.
-    Don't forget to remove the puzzle.
--->
-<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" id="rt-without-atom" version="2.0">
+<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" id="rt-and-atoms" version="2.0">
   <xsl:output encoding="UTF-8" method="xml"/>
   <xsl:template match="/program/errors">
     <xsl:copy>
       <xsl:apply-templates select="node()|@*"/>
-      <xsl:if test="/program/metas/meta[head='rt']">
-        <xsl:if test="not(//o[@atom])">
+      <xsl:if test="//o[@atom]">
+        <xsl:if test="not(/program/metas/meta[head='rt'])">
+          <xsl:element name="error">
+            <xsl:attribute name="check">
+              <xsl:text>rt-and-atoms</xsl:text>
+            </xsl:attribute>
+            <xsl:attribute name="line">
+              <xsl:value-of select="@line"/>
+            </xsl:attribute>
+            <xsl:attribute name="severity">
+              <xsl:text>error</xsl:text>
+            </xsl:attribute>
+            <xsl:text>Using atoms without +rt meta</xsl:text>
```

Comment:
@c71n93 let's make this message more informative, like "Using atoms without +rt meta is prohibited"

Reply:
@maxonfjvipon done